From 92a6ec361dce668cbe6389f018e32ae08d2307de Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 21 Jun 2018 23:11:44 +0000
Subject: [PATCH] Add Clear Linux docker integration for kata containers

Set kata containers as the default docker runtime on Clear Linux.
---
 clearlinux.conf            | 7 +++++++
 docker-set-runtime.service | 8 ++++++++
 set-docker-default-runtime | 4 ++++
 3 files changed, 19 insertions(+)
 create mode 100644 clearlinux.conf
 create mode 100644 docker-set-runtime.service
 create mode 100755 set-docker-default-runtime

diff --git a/clearlinux.conf b/clearlinux.conf
new file mode 100644
index 0000000..869d29d
--- /dev/null
+++ b/clearlinux.conf
@@ -0,0 +1,7 @@
+[Unit]
+Wants=docker-set-runtime.service
+After=docker-set-runtime.service
+
+[Service]
+Environment="DOCKER_EXTRA_RUNTIMES=--add-runtime cc-runtime=/usr/bin/cc-runtime"
+ExecStart=/usr/bin/dockerd $DOCKER_EXTRA_RUNTIMES $DOCKER_DEFAULT_RUNTIME --storage-driver=overlay2
diff --git a/docker-set-runtime.service b/docker-set-runtime.service
new file mode 100644
index 0000000..0cc4fdd
--- /dev/null
+++ b/docker-set-runtime.service
@@ -0,0 +1,8 @@
+[Unit]
+Description=Set Default Docker Runtime
+ConditionPathExists=!/etc/systemd/system/docker.service.d/50-runtime.conf
+ConditionPathExists=/dev/kvm
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/set-docker-default-runtime
diff --git a/set-docker-default-runtime b/set-docker-default-runtime
new file mode 100755
index 0000000..8934b13
--- /dev/null
+++ b/set-docker-default-runtime
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+mkdir -p /etc/systemd/system/docker.service.d/
+echo -e '[Service]\nEnvironment="DOCKER_DEFAULT_RUNTIME=--default-runtime cc-runtime"' > /etc/systemd/system/docker.service.d/50-runtime.conf
-- 
2.17.1

